src/NHSD.BuyingCatalogue.Organisations.Api/Services/MailKitEmailService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using MailKit;
+using NHSD.BuyingCatalogue.Organisations.Api.Settings;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    /// <summary>
+    /// A service for sending e-mails using MailKit.
+    /// </summary>
+    internal sealed class MailKitEmailService : IEmailService
+    {
+        private readonly IMailTransport _client;
+        private readonly SmtpSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MailKitEmailService"/> class
+        /// using the provided <paramref name="client"/> and <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="client">The mail transport to use to send e-mail.</param>
+        /// <param name="settings">The SMTP configuration.</param>
+        public MailKitEmailService(IMailTransport client, SmtpSettings settings)
+        {
+            _client = client;
+
+            // TODO: confirm this is acceptable for us if using SSL
+            // (as we will be using a trusted server that is not
+            // user-configurable this should be fine, providing the
+            // root CA is trusted by the host system)
+            client.ServerCertificateValidationCallback = (s, c, h, e) => true;
+
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// Sends an e-mail message asynchronously using the
+        /// provided <see cref="IMailTransport"/> instance.
+        /// </summary>
+        /// <param name="emailMessage">The e-mail message to send asynchronously.</param>
+        /// <returns>An asynchronous task context.</returns>
+        public async Task SendEmailAsync(EmailMessage emailMessage)
+        {
+            await _client.ConnectAsync(_settings.Host, _settings.Port, _settings.UseSsl);
```

Comment:
Is IMailTransport disposable? I ask as I think it would mean this service would have to be disposable as well.

Reply:
`IMailTransport` is disposable but we shouldn't need to implement `IDisposable` here because `MailKitEmailService` does not own the `SmtpClient` instance – the DI container does so it is responsible for disposing of it.

One can argue having `IMailTransport` derive from `IDisposable` is a bit of a smell – as the use of unmanaged resources depends on the implementation not the abstraction. Unfortunately, it's not something we have control of in this case.